crates/bevy_audio/Cargo.toml

```diff
@@ -21,6 +21,9 @@ anyhow = "1.0.4"
 rodio = { version = "0.16", default-features = false }
 parking_lot = "0.12.1"
 
+[target.'cfg(target_os = "android")'.dependencies]
+oboe = { version = "0.4", features = ["shared-stdcxx"] }
```

Comment:
I vaguely recall this being discussed before but can't remember why it's necessary to use `shared-stdcxx` here?

It seems surprising to force the use of libc++_shared.so here and since there's currently no standard convention with rust crates for configuring whether to statically or dynamically link with stdc++ I'd worry that this could lead to trouble down the line if you have some things statically linking and others dynamically linking.

... okey, poking around I see this got discussed here before when I was surprised by the same thing: https://github.com/bevyengine/bevy/issues/86#issuecomment-1336549374

@MarijnS95 was able to confirm that `cargo apk` will recognise if a crate dynamically links with libc++_shared.so and will automatically copy the library from the NDK into the APK that's built, but I still can't say I understand why there's any need to choose dynamic linking here, and I'd still be concerned about having inconsistency between crates.

Reply:
I added a feature to have it by default but to be able to not enable it